### PR TITLE
feat: newsletter improvment

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes https://github.com/gravitee-io/issues/issues/4692

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ export { GvTree } from './molecules/gv-tree';
 export { GvDocumentation } from './organisms/gv-documentation';
 export { GvHeader } from './organisms/gv-header';
 export { GvMenu } from './organisms/gv-menu';
+export { GvNewsletterSubscriptionPage } from './organisms/gv-newsletter-subscription-page';
```
Comment (gaetanmaisse) [inline src/index.js:53]: Maybe we can go with `GvNewsletterSubscription`? As there is no more `:root` CSS I'm not sure the `Page` term is still right

File: assets/icons/thirdparty/twitter.svg [hunk 2]
```diff
@@ -1,7 +1,3 @@
...
-    <defs>
-        <style>.cls-2{fill:#1da1f2;}</style>
-    </defs>
-    <path class="cls-2"
-          d="M153.62,301.59c94.34,0,145.94-78.16,145.94-145.94,0-2.22,0-4.43-.15-6.63A104.36,104.36,0,0,0,325,122.47a102.38,102.38,0,0,1-29.46,8.07,51.47,51.47,0,0,0,22.55-28.37,102.79,102.79,0,0,1-32.57,12.45,51.34,51.34,0,0,0-87.41,46.78A145.62,145.62,0,0,1,92.4,107.81a51.33,51.33,0,0,0,15.88,68.47A50.91,50.91,0,0,1,85,169.86c0,.21,0,.43,0,.65a51.31,51.31,0,0,0,41.15,50.28,51.21,51.21,0,0,1-23.16.88,51.35,51.35,0,0,0,47.92,35.62,102.92,102.92,0,0,1-63.7,22A104.41,104.41,0,0,1,75,278.55a145.21,145.21,0,0,0,78.62,23"/>
+  <path fill="black" d="M153.62,301.59c94.34,0,145.94-78.16,145.94-145.94,0-2.22,0-4.43-.15-6.63A104.36,104.36,0,0,0,325,122.47a102.38,102.38,0,0,1-29.46,8.07,51.47,51.47,0,0,0,22.55-28.37,102.79,102.79,0,0,1-32.57,12.45,51.34,51.34,0,0,0-87.41,46.78A145.62,145.62,0,0,1,92.4,107.81a51.33,51.33,0,0,0,15.88,68.47A50.91,50.91,0,0,1,85,169.86c0,.21,0,.43,0,.65a51.31,51.31,0,0,0,41.15,50.28,51.21,51.21,0,0,1-23.16.88,51.35,51.35,0,0,0,47.92,35.62,102.92,102.92,0,0,1-63.7,22A104.41,104.41,0,0,1,75,278.55a145.21,145.21,0,0,0,78.62,23"/>
```
Comment (gaetanmaisse) [inline assets/icons/thirdparty/twitter.svg:2]: What happens to the Twitter icon?

File: src/index.js (hunk 1 above)
Comment (gaetanmaisse) [reply src/index.js:53]: --> It looks like there is already a GvNewsletterSubscription component in APIM, maybe it feels better to rename it to something like GvNewsletterReminder and use GvNewsletterSubscription for the new web component

File: assets/icons/thirdparty/twitter.svg (hunk 2 above)
Comment (ytvnr) [reply assets/icons/thirdparty/twitter.svg:2]: it was colored with brand twitter blue so it was not overridable with css

so we make it customizable

File: src/index.js (hunk 1 above)
Comment (gcusnieux) [reply src/index.js:53]: 👍 agree

File: src/organisms/gv-newsletter-subscription-page.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+      header: { type: String },
+      emailPlaceholder: { type: String, attribute: 'email-placeholder' },
+      subscribeLabel: { type: String, attribute: 'subscribe-label' },
+      skipLabel: { type: String, attribute: 'skip-label' },
+      taglines: { type: Array },
+      email: { type: String },
+      emailDisabled: { type: Boolean },
+      githubUrl: { type: String },
+      twitterUrl: { type: String },
+      linkedinUrl: { type: String },
```
Comment (gcusnieux) [inline src/organisms/gv-newsletter-subscription-page.js]: You can use attribute to specify name property in kebab-case like others attributes.
Also I prefer simple attribute like label, placeholder, disabled.

File: src/organisms/gv-newsletter-subscription-page.js [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+          display: flex;
+          flex-direction: row;
+          justify-content: flex-end;
+          margin-bottom: -15px;
+        }
+
+        .social-link {
+          --gv-icon--c: var(--gv-theme-color-light, #86c3d0);
+          --gv-icon--s: 32px;
+          margin-right: -8px;
```
Comment (gcusnieux) [inline src/organisms/gv-newsletter-subscription-page.js]: it's not very elegant, what did you want to do?

File: src/organisms/gv-newsletter-subscription-page.js [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+      this.email = detail;
+      const emailInput = this.shadowRoot.querySelector('#email');
+      const emailSubmit = this.shadowRoot.querySelector('gv-button#email-submit');
+      if (emailInput.invalid && !this.emailDisabled) {
+        emailSubmit.setAttribute('disabled', true);
+      }
+      else {
+        emailSubmit.removeAttribute('disabled');
+      }
+    });
```
Comment (gcusnieux) [inline src/organisms/gv-newsletter-subscription-page.js]: You can use lit-element lifecycle and use a property for disable the submit button

File: src/index.js (hunk 1 above)
Comment (ytvnr) [reply src/index.js]: it's fixed :)

File: src/organisms/gv-newsletter-subscription.js [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+      twitterUrl: { type: String },
+      linkedinUrl: { type: String },
+    };
+  }
+
+  static get styles () {
+    return [
+      // language=css
+      css`
+        .newsletter-container {
```
Comment (gcusnieux) [inline src/organisms/gv-newsletter-subscription.js:60]: On each component we initialize two default properties

`:host {
  box-sizing: border-box;
  margin: 0.2rem;
}`

**Review by gcusnieux — APPROVED** [review]

**Review by gaetanmaisse — APPROVED** [review]

File: src/organisms/gv-newsletter-subscription-page.js (hunk 3 above)
Comment (ytvnr) [reply src/organisms/gv-newsletter-subscription-page.js]: done

**Review by aelamrani — APPROVED** [review]

File: src/organisms/gv-newsletter-subscription-page.js (hunk 3 above)
Comment (gcusnieux) [reply src/organisms/gv-newsletter-subscription-page.js]: Sorry @ytvnr, my suggest was :

Comment (ytvnr) [reply src/organisms/gv-newsletter-subscription-page.js]: and for placeholder ? can we keep 'emailPlaceholder' or you prefer 'placeholder' ?

as @aelamrani tolds me, only 'placeholder' does not give a clue on which we are talking about

Comment (aelamrani) [reply src/organisms/gv-newsletter-subscription-page.js]: I was just quoting Guillaume (at least I thought it was his previous message here) but seems he is fine with this

**Pull request merged**